internal/git/commands.go

```diff
@@ -924,101 +941,90 @@ func NewUnmergedStage(value int) (UnmergedStage, error) {
 	return 0, fmt.Errorf("unknown stage ID: %q", value)
 }
 
-func ParseVerboseBranchesOutput(output string) (gitdomain.BranchInfos, Option[gitdomain.LocalBranchName]) {
-	result := gitdomain.BranchInfos{}
-	spaceRE := regexp.MustCompile(" +")
+type branchesQueryResult struct {
+	BranchName     string
```

Comment:
Based on the documentation you added for the query (thanks for that btw), I think this could be either `branch-name` (which would be a `gitdomain.LocalBranchName`) or `origin/branch-name` (which would be a `gitdomain.RemoteBranchName`). If this is correct, let's indicate this via the type system. 

```suggestion
	BranchName     gitdomain.BranchName
```

Advantages:

- this documents that the content can have two different meanings
- this forces the user to check the string before assuming it's a local or remote branch
- code to perform these checks already exists

Reply:
Done.